tests/e2e/replication/show_while_creating_invalid_state.py

```diff
@@ -117,7 +118,283 @@ def test_show_replicas(connection):
+
+    data_directory1 = tempfile.TemporaryDirectory()
+    data_directory2 = tempfile.TemporaryDirectory()
+    CONFIGURATION = {
+        "main_1": {
+            "args": ["--bolt-port", "7687", "--log-level=TRACE"],
+            "log_file": "main1.log",
+            "setup_queries": [],
+            "data_directory": f"{data_directory1.name}",
+        },
+        "main_2": {
+            "args": ["--bolt-port", "7688", "--log-level=TRACE"],
+            "log_file": "main2.log",
+            "setup_queries": [],
+            "data_directory": f"{data_directory2.name}",
+        },
+    }
+
+    interactive_mg_runner.start_all(CONFIGURATION)
+    cursor_1 = connection(7687, "main_1").cursor()
+    cursor_2 = connection(7688, "main_2").cursor()
+
+    assert execute_and_fetch_all(cursor_1, "SHOW REPLICATION ROLE;")[0][0] == "main"
+    assert execute_and_fetch_all(cursor_2, "SHOW REPLICATION ROLE;")[0][0] == "main"
+
+
+def test_basic_recovery_when_replica_is_kill_when_main_is_down(connection):
+    # Goal of this test is to check the recovery of main.
+    # 0/ We start all replicas manually: we want to be able to kill them ourselves without relying on external tooling to kill processes.
+    # 1/ We check that all replicas have the correct state: they should all be ready.
+    # 2/ We kill main then kill a replica.
+    # 3/ We re-start main: it should be able to restart.
+    # 4/ Check status of replica: replica_2 is invalid.
+
+    data_directory = tempfile.TemporaryDirectory()
+    CONFIGURATION = {
+        "replica_1": {
+            "args": ["--bolt-port", "7688", "--log-level=TRACE"],
+            "log_file": "replica1.log",
+            "setup_queries": ["SET REPLICATION ROLE TO REPLICA WITH PORT 10001;"],
+        },
+        "replica_2": {
+            "args": ["--bolt-port", "7689", "--log-level=TRACE"],
+            "log_file": "replica2.log",
+            "setup_queries": ["SET REPLICATION ROLE TO REPLICA WITH PORT 10002;"],
+        },
+        "main": {
+            "args": ["--bolt-port", "7687", "--log-level=TRACE", "--storage-recover-on-startup=true"],
+            "log_file": "main.log",
+            "setup_queries": [],
+            "data_directory": f"{data_directory.name}",
+        },
+    }
+
+    interactive_mg_runner.start_all(CONFIGURATION)
+
+    # We want to execute manually and not via the configuration, otherwise re-starting main would also execute these registration.
+    interactive_mg_runner.MEMGRAPH_INSTANCES["main"].query("REGISTER REPLICA replica_1 SYNC TO '127.0.0.1:10001';")
+    interactive_mg_runner.MEMGRAPH_INSTANCES["main"].query("REGISTER REPLICA replica_2 SYNC TO '127.0.0.1:10002';")
+
+    # 1/
+    expected_data = {
+        ("replica_1", "127.0.0.1:10001", "sync", 0, 0, "ready"),
+        ("replica_2", "127.0.0.1:10002", "sync", 0, 0, "ready"),
+    }
+    actual_data = set(interactive_mg_runner.MEMGRAPH_INSTANCES["main"].query("SHOW REPLICAS;"))
+
+    assert actual_data == expected_data
```

Comment:
same as above - we should have some retrying logic that gives the test some time for nodes to connect and set things up without relying on sleeps, which will cause both flakes and slower tests than necessary over time.